source/framework/vx_copy_swap.c

```diff
@@ -0,0 +1,503 @@
+                    {
+                        if ((othernode->parameters[k] == first) &&
+                            ((vx_enum)VX_INPUT == othernode->kernel->signature.directions[k]))
+                        {
+                            /* This is not allowed */
+                            VX_PRINT(VX_ZONE_ERROR, "Move nodes must not have their first parameter connected to another node's input\n");
+                            status = (vx_status)VX_FAILURE;
+                            break;
+                        }
+                    }
+                }
+            }
+        }
+    }
+    /* Copy and move node removal - Output nodes.
+     * When ever we remove an output node, another
+     * node may become an output node, so we iterate
+     * until no more are found.
+     */
+    do
+    {
+        node_removed = (vx_bool)vx_false_e;
+        for (i = 0; (i < num_copy_move_nodes) && ((vx_status)VX_SUCCESS == status); ++i)
+        {
+            vx_node node = graph->nodes[copy_move_indices[i]];
+            vx_reference first = node->parameters[0];
+            vx_reference second = node->parameters[1];
+            vx_bool removable = first->is_virtual;
+            if ((vx_bool)vx_false_e == second->is_virtual)
+            {
+                /* Copy nodes may only be removed if the input is not connected to another copy node input */
+                if ((vx_enum)VX_KERNEL_COPY == node->kernel->enumeration)
+                {
+                    vx_uint32 j;
+                    for (j = 0; j < num_copy_move_nodes; ++j)
+                    {
+                        if (i != j)
+                        {
+                            if (first == graph->nodes[copy_move_indices[j]]->parameters[0])
+                            {
+                                removable = (vx_bool)vx_false_e;
+                                break;
+                            }
+                        }
+                    }
+                }
+                if (((vx_bool)vx_true_e == removable) &&
+                    ((vx_bool)vx_true_e == ownCopyMoveReplaceInputCompatible(first, second)))
+                {
+                    /* we can remove this node, propagating the output reference */
+                    status = ownCopyMoveRemoveNode(graph, copy_move_indices[i], first, second);
+                    if ((vx_status)VX_SUCCESS == status)
+                    {
+                        node_removed = (vx_bool)vx_true_e;
+                        /* and re-make node list list */
+                        num_copy_move_nodes = makeCopyMoveNodeList(graph, copy_move_indices);
+                        /* Now process any graph parameters connected to the output */
+                        ownReassignGraphParameter(graph, node, 1, second, (vx_enum)VX_OUTPUT);
+                        /* Re-start the loop */
+                        break;
+                    }
+                    else
```

Comment:
Shall we remove the empty else clause here since this is a simple "if" condition above it and not an "if else"?  Just flagging as a code coverage gap which may be unnecessary.

Reply:
empty else removed